### DISCUSSION TITLE
Remove v1alpha1 version of resources which are available in v1beta1

### DISCUSSION THREAD

**Discussion by afrittoli** [tektoncd/pipeline#4633] [General]
It's been almost an year since [Tekton Pipelines Beta release v0.11.0 "Ragdoll Norby"](https://github.com/tektoncd/pipeline/releases/tag/v0.11.0) where we introduce the `v1beta1` version of the following CRDs:

- `Task`
- `ClusterTask`
- `Pipeline`
- `TaskRun`
- `PipelineRun`

I don't recall us ever officially deprecating the `v1alpha` version of these five CRDs, I think we might have assumed it was implied by releasing their `v1beta` version. In any case, since they're alpha, the [API compatibility policy](https://github.com/tektoncd/pipeline/blob/main/api_compatibility_policy.md) allows us to remove them with one release of warning time.

I would propose the we:

- include the five `v1alpha1` resources in the deprecation table in v0.34.x
- stop serving / supporting those five `v1alpha1` resources in v0.35.x

@tektoncd/core-maintainers

**Comment by vdemeester** [comment]
I would love that 👼🏼 😝 
@tektoncd/cli-maintainers that would be another reason for droping v1alpha1 support in the next releases of `tkn` 😛

**Comment by ghost** [comment]
This sounds like a great idea to me. This would allow us to remove the conversion webhook code that stores v1beta1 fields as annotations on v1alpha1 resources.

**Comment by pritidesai** [comment]
yup, definitely, creating `V1` will be a little simpler without those alpha resources.
